File: mm-php/src/io/metamessage/mm/MimeWire.php
<?php

namespace io\metamessage\mm;

class MimeWire {
    private static array $STR_TO_MIME = [
        'image/jpeg' => 1,
        'image/jpg' => 1,
        'image/png' => 2,
        'image/gif' => 3,
        'image/webp' => 4,
        'image/svg+xml' => 5,
        'image/avif' => 6,
        'image/bmp' => 7,
        'image/x-icon' => 8,
        'image/tiff' => 9,
        'image/heic' => 10,
        'image/heif' => 11,
        'text/plain' => 12,
        'text/html' => 13,
        'text/css' => 14,
        'text/javascript' => 15,
        'application/javascript' => 15,
        'application/json' => 16,
        'text/csv' => 17,
        'text/markdown' => 18,
        'application/pdf' => 19,
        'application/zip' => 20,
        'application/gzip' => 21,
        'application/x-tar' => 22,
        'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet' => 23,
        'application/vnd.openxmlformats-officedocument.wordprocessingml.document' => 24,
        'application/octet-stream' => 25,
        'video/mp4' => 26,
        'video/webm' => 27,
        'video/mov' => 28,
        'audio/mpeg' => 29,
        'audio/wav' => 30,
        'audio/flac' => 31,
        'font/woff2' => 32,
        'font/ttf' => 33
    ];

    public static function parse(?string $s): int {
        if (empty($s)) {
            return 0;
        }
        $key = strtolower(trim($s));
        return self::$STR_TO_MIME[$key] ?? 0;
    }

    public static function toString(int $code): string {
        return match($code) {
            1 => 'image/jpeg',
            2 => 'image/png',
            3 => 'image/gif',
            4 => 'image/webp',
            5 => 'image/svg+xml',
            6 => 'image/avif',
            7 => 'image/bmp',
            8 => 'image/x-icon',
            9 => 'image/tiff',
            10 => 'image/heic',
            11 => 'image/heif',
            12 => 'text/plain',
            13 => 'text/html',
            14 => 'text/css',
            15 => 'text/javascript',
            16 => 'application/json',
            17 => 'text/csv',
            18 => 'text/markdown',
            19 => 'application/pdf',
            20 => 'application/zip',
            21 => 'application/gzip',
            22 => 'application/x-tar',
            23 => 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
            24 => 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
            25 => 'application/octet-stream',
            26 => 'video/mp4',
            27 => 'video/webm',
            28 => 'video/mov',
            29 => 'audio/mpeg',
            30 => 'audio/wav',
            31 => 'audio/flac',
            32 => 'font/woff2',
            33 => 'font/ttf',
            default => 'unknown'
        };
    }
}

File: mm-java/src/main/java/io/metamessage/mm/MimeWire.java
package io.github.metamessage.mm;

import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

final class MimeWire {
    private static final Map<String, Integer> STR_TO_MIME = new HashMap<>();

    static {
        put("image/jpeg", 1);
        put("image/jpg", 1);
        put("image/png", 2);
        put("image/gif", 3);
        put("image/webp", 4);
        put("image/svg+xml", 5);
        put("image/avif", 6);
        put("image/bmp", 7);
        put("image/x-icon", 8);
        put("image/tiff", 9);
        put("image/heic", 10);
        put("image/heif", 11);
        put("text/plain", 12);
        put("text/html", 13);
        put("text/css", 14);
        put("text/javascript", 15);
        put("application/javascript", 15);
        put("application/json", 16);
        put("text/csv", 17);
        put("text/markdown", 18);
        put("application/pdf", 19);
        put("application/zip", 20);
        put("application/gzip", 21);
        put("application/x-tar", 22);
        put("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", 23);
        put("application/vnd.openxmlformats-officedocument.wordprocessingml.document", 24);
        put("application/octet-stream", 25);
        put("video/mp4", 26);
        put("video/webm", 27);
        put("video/mov", 28);
        put("audio/mpeg", 29);
        put("audio/wav", 30);
        put("audio/flac", 31);
        put("font/woff2", 32);
        put("font/ttf", 33);
    }

    private static void put(String k, int v) {
        STR_TO_MIME.put(k.toLowerCase(Locale.ROOT), v);
    }

    static int parse(String s) {
        if (s == null || s.isBlank()) {
            return 0;
        }
        Integer m = STR_TO_MIME.get(s.trim().toLowerCase(Locale.ROOT));
        return m != null ? m : 0;
    }

    static String toString(int code) {
        return switch (code) {
            case 1 -> "image/jpeg";
            case 2 -> "image/png";
            case 3 -> "image/gif";
            case 4 -> "image/webp";
            case 5 -> "image/svg+xml";
            case 6 -> "image/avif";
            case 7 -> "image/bmp";
            case 8 -> "image/x-icon";
            case 9 -> "image/tiff";
            case 10 -> "image/heic";
            case 11 -> "image/heif";
            case 12 -> "text/plain";
            case 13 -> "text/html";
            case 14 -> "text/css";
            case 15 -> "text/javascript";
            case 16 -> "application/json";
            case 17 -> "text/csv";
            case 18 -> "text/markdown";
            case 19 -> "application/pdf";
            case 20 -> "application/zip";
            case 21 -> "application/gzip";
            case 22 -> "application/x-tar";
            case 23 -> "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
            case 24 -> "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
            case 25 -> "application/octet-stream";
            case 26 -> "video/mp4";
            case 27 -> "video/webm";
            case 28 -> "video/mov";
            case 29 -> "audio/mpeg";
            case 30 -> "audio/wav";
            case 31 -> "audio/flac";
            case 32 -> "font/woff2";
            case 33 -> "font/ttf";
            default -> "unknown";
        };
    }

    private MimeWire() {}
}
